### PR TITLE
chore(action): use new method to declare schemas

### PR DESCRIPTION
**Pull request by sileht**
This allows all schema to use partial_validation and avoid any
bug/mistake with schema that use mutable object as default (eg: []).

Change-Id: I68e9ce851b01ee3a38cc367a481064bb45e37c82

### CODE REVIEW

File: mergify_engine/actions/assign.py [hunk 1]
```diff
@@ -28,19 +28,23 @@
...
         | actions.ActionFlag.ALWAYS_RUN
     )
 
+    @classmethod
+    def get_config_schema(
+        cls, partial_validation: bool
+    ) -> typing.Dict[typing.Any, typing.Any]:
+        return {
+            # NOTE: "users" is deprecated, but kept as legacy code for old config
+            voluptuous.Required("users", default=[]): [types.Jinja2],
```
Comment (jd) [inline mergify_engine/actions/assign.py:43]: It's not obvious, but you can actually use `default=list` and voluptuous will see it's a callable and call it.

That might be a simpler change?

File: mergify_engine/actions/__init__.py [hunk 2]
```diff
@@ -109,22 +115,16 @@ def get_schema(cls, partial_validation: bool = False) -> ActionSchema:
...
     def command_to_config(string: str) -> typing.Dict[str, typing.Any]:
         """Convert string to dict config"""
         return {}
 
+    @abc.abstractmethod
     async def run(
         self, ctxt: context.Context, rule: "rules.EvaluatedRule"
     ) -> check_api.Result:  # pragma: no cover
-        pass
+        ...
```
Comment (jd) [inline mergify_engine/actions/__init__.py:127]: Looks unrelated?

File: mergify_engine/actions/assign.py (hunk 1 above)
Comment (sileht) [reply mergify_engine/actions/assign.py:43]: https://github.com/Mergifyio/mergify-engine/pull/3150
